app/src/main/java/org/p2p/wallet/moonpay/MoonpayModule.kt

```diff
@@ -0,0 +1,57 @@
+package org.p2p.wallet.moonpay
+
+import org.koin.core.module.dsl.factoryOf
+import org.koin.core.qualifier.named
+import org.koin.dsl.module
+import org.p2p.wallet.BuildConfig
+import org.p2p.wallet.common.di.InjectionModule
+import org.p2p.wallet.home.HomeModule
+import org.p2p.wallet.moonpay.api.MoonpayApi
+import org.p2p.wallet.moonpay.api.MoonpayUrlBuilder
+import org.p2p.wallet.moonpay.repository.MoonpayApiMapper
+import org.p2p.wallet.moonpay.repository.MoonpayBuyRemoteRepository
+import org.p2p.wallet.moonpay.repository.MoonpayBuyRepository
+import org.p2p.wallet.moonpay.repository.NewMoonpayBuyRemoteRepository
+import org.p2p.wallet.moonpay.repository.NewMoonpayBuyRepository
+import org.p2p.wallet.moonpay.repository.sell.MoonpaySellRemoteRepository
+import org.p2p.wallet.moonpay.repository.sell.MoonpaySellRepository
+import retrofit2.Retrofit
+
+object MoonpayModule : InjectionModule {
+    override fun create() = module {
+        factoryOf(::MoonpayApiMapper)
+
+        factory<MoonpayApi> {
+            val retrofit = get<Retrofit>(named(HomeModule.MOONPAY_QUALIFIER))
+            retrofit.create(MoonpayApi::class.java)
+        }
+
+        factory<MoonpayBuyRepository> {
+            val apiKey = BuildConfig.moonpayKey
+            MoonpayBuyRemoteRepository(
+                api = get(),
+                moonpayApiKey = apiKey,
+                mapper = get()
+            )
+        }
+        factory<NewMoonpayBuyRepository> {
+            val apiKey = BuildConfig.moonpayKey
+            NewMoonpayBuyRemoteRepository(get(), apiKey, get())
```

Comment:
`bind` would be more obvious for me. 
And why you are not using BuildConfig from the repository? 
You avoid using `get`s here

Reply:
i saw we are injecting the key in DI and did so too. 
will move.